src/pip/_internal/download.py

```diff
@@ -140,13 +145,31 @@ def __init__(self, prompting=True):
         self.passwords = {}
 
     def __call__(self, req):
```

Comment:
I think it would be helpful to put `__call__()` after `_get_url_and_credentials()` since it depends on that method, so you can read it from top to bottom more easily.